packages/amplify-graphql-relational-transformer/src/schema.ts

```diff
@@ -120,179 +137,235 @@ export function ensureHasOneConnectionField(config: HasOneDirectiveConfiguration
-    ctx.output.putType(updateFilterConnectionInputWithConnectionField(conditionInput, connectionAttributeName));
+    ctx.output.putType({
+      ...conditionInput,
+      fields: [
+        ...filterInput.fields!,
+        ...getFilterConnectionInputFieldsWithConnectionField([...conditionInput.fields!], connectionAttributeName),
+      ],
+    });
   }
   config.connectionFields.push(connectionAttributeName);
-}
+};
 
 /**
  * If the related type is a hasOne relationship, this creates a hasOne relation going the other way
  *    but using the same foreign key name as the hasOne model
  * If the related type is a hasMany relationship, this function sets the foreign key name to the name of the hasMany foreign key
  *    but does not add additional fields as this will be handled by the hasMany directive
  */
-export function ensureBelongsToConnectionField(config: BelongsToDirectiveConfiguration, ctx: TransformerContextProvider) {
+export const ensureBelongsToConnectionField = (config: BelongsToDirectiveConfiguration, ctx: TransformerContextProvider): void => {
   const { relationType, relatedType, relatedField } = config;
   if (relationType === 'hasOne') {
     ensureHasOneConnectionField(config, ctx);
   } else {
     // hasMany
     config.connectionFields.push(getConnectionAttributeName(relatedType.name.value, relatedField.name.value));
+    config.connectionFields.push(...getSortKeyFieldNames(relatedType));
   }
-}
+};
 
-export function ensureHasManyConnectionField(
+/**
+ * ensureHasManyConnectionField
+ */
+export const ensureHasManyConnectionField = (
   config: HasManyDirectiveConfiguration | ManyToManyDirectiveConfiguration,
   ctx: TransformerContextProvider,
-) {
-  const { field, fieldNodes, object, relatedType } = config;
+): void => {
+  const {
+    field, fieldNodes, object, relatedType,
+  } = config;
 
   // If fields were explicitly provided to the directive, there is nothing else to do here.
   if (fieldNodes.length > 0) {
     return;
   }
 
+  let connectionFieldName = 'id';
+  const sortKeyFields = getSortKeyFields(ctx, object);
+
+  object.fields!.forEach(objectField => {
+    objectField.directives!.forEach(directive => {
+      if (directive.name.value === 'primaryKey') {
+        connectionFieldName = objectField.name.value;
+      }
+    });
+  });
+
+  config.connectionFields.push(connectionFieldName, ...sortKeyFields.map(it => it.name.value));
+
   const connectionAttributeName = getConnectionAttributeName(object.name.value, field.name.value);
 
   const relatedTypeObject = ctx.output.getType(relatedType.name.value) as ObjectTypeDefinitionNode;
   if (relatedTypeObject) {
-    ctx.output.putType(updateTypeWithConnectionField(relatedTypeObject, connectionAttributeName, isNonNullType(field.type)));
+    const updatedFields = [...relatedTypeObject.fields!];
+    updatedFields.push(...getTypeFieldsWithConnectionField(updatedFields, connectionAttributeName, isNonNullType(field.type)));
+    sortKeyFields.forEach(it => {
```

Comment:
@alharris-at this is one the actual change, adding the sortKeyFields from the hasMany object to the belongsTo object for proper lookup

Reply:
If we add the sortKeyFields from the hasMany object w/o any renaming, there might be duplicates if the belongsTo model has same field names as sortKeyFields right? I believe that is the reason why we generate a `{hasManyModelName}{belongsToModelName}Id` field versus using the direct primary key field name from the hasMany model.

Reply:
That is right, will update to use same naming structure for sortKeyFields as well